GitUtils.ps1

```diff
@@ -349,10 +372,21 @@ function Add-SshKey() {
             }
         }
     } else {
-        $sshAdd = Get-Command ssh-add -TotalCount 1 -ErrorAction SilentlyContinue
-        $sshAdd = if ($sshAdd) {$sshAdd} else {Find-Ssh('ssh-add')}
+        $agentPid = (Get-Item Env:SSH_AGENT_PID -ErrorAction SilentlyContinue).Value
+        $sshAdd = '';
+        if($agentPid){
+            $agentPath = (Get-Process -Id $agentPid -FileVersionInfo -ErrorAction SilentlyContinue).Filename
+            if($agentPath){
+                $sshPath = Split-Path ($agentPath) -Parent
+            }
+            if($sshPath){
+                $sshAdd = Get-Command (Join-Path $sshPath 'ssh-add')
+            }
+        }
+        if(!$sshAdd){
+            $sshAdd = Find-Ssh('ssh-add')
+        }

```

Comment:
This feels overcomplicated- if we've got a running ssh-agent, `Get-Command ssh-add` or `Find-Ssh('ssh-add')` should return the matching ssh-add so just calling `ssh-add` should work. If not, maybe the logic to find the correct binary for the agent should be in Find-Ssh?